### PR TITLE
Improve error message for timeout errors

### LINKED ISSUE #6074: Provide a better user message on connection timeout errors when querying events API
image

The above could be improve with a better more human readable error messages like `Unable to complete the request due to timeout issues, try again later in a few minutes`

### PR DESCRIPTION
**Pull request by pierrebeaucamp**
With this change `httplib` will attempt to identify timeout errors thrown by `net/http`, specifically when using a custom transport.

I originally added this error handling to the `SearchEvents` method in `lib/auth/ctl.go` since that is the specific instance reported in #6074. However, having a one-off special error handling case felt wrong, so I moved it to the more generic `ConvertResponse` function from `httplib`. This in turn might be _too_ broad, so please let me know if we don't want that in this central place as it might affect error handling for other components as well.

This fixes #6074

### CODE REVIEW

File: lib/httplib/httplib.go [hunk 1]
```diff
@@ -126,6 +136,13 @@ func ConvertResponse(re *roundtrip.Response, err error) (*roundtrip.Response, er
 		if uerr, ok := err.(*url.Error); ok && uerr != nil && uerr.Err != nil {
 			return nil, trace.ConnectionProblem(uerr.Err, uerr.Error())
 		}
+		// Since the original timeout error is not exported, using `errors.Is` with our custom error will always
+		// result in this check being false. Unwrapping and comparing the error string works though.
+		if uerr := errors.Unwrap(err); uerr != nil && uerr.Error() == errorTransportTimeout.Error() {
```
Comment (awly) [inline lib/httplib/httplib.go:141]:
```suggestion
		if nerr, ok := errors.Unwrap(err).(net.Error); ok && nerr.Timeout() {
```

File: lib/httplib/httplib.go [hunk 2]
```diff
@@ -36,6 +37,15 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+// timeoutMessage is a generic "timeout" error message that is displayed as a more user-friendly alternative to
+// the timeout errors returned by net/http
+const timeoutMessage = "Unable to complete the request due to a timeout, please try again in a few minutes"
```
Comment (awly) [inline lib/httplib/httplib.go:42]:
```suggestion
const timeoutMessage = "unable to complete the request due to a timeout, please try again in a few minutes"
```

File: lib/httplib/httplib.go [hunk 3]
```diff
@@ -36,6 +37,15 @@ import (
...
+
+// errorTransportTimeout is the error returned by net/http in case of a timeout. As of Go 1.16, it is not
+// exported by the standard library, therefore we replicate it here. See:
+// https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/net/http/transport.go#L2505
+var errorTransportTimeout = errors.New("net/http: timeout awaiting response headers")
```
Comment (awly) [inline lib/httplib/httplib.go:47]:
```suggestion
```

File: lib/httplib/httplib.go [hunk 4]
```diff
@@ -126,6 +136,13 @@ func ConvertResponse(re *roundtrip.Response, err error) (*roundtrip.Response, er
 		if uerr, ok := err.(*url.Error); ok && uerr != nil && uerr.Err != nil {
 			return nil, trace.ConnectionProblem(uerr.Err, uerr.Error())
 		}
+		// Since the original timeout error is not exported, using `errors.Is` with our custom error will always
+		// result in this check being false. Unwrapping and comparing the error string works though.
```
Comment (awly) [inline lib/httplib/httplib.go:140]:
```suggestion
```

File: lib/httplib/httplib.go (hunk 1 above)
Comment (pierrebeaucamp) [reply lib/httplib/httplib.go:141]: TIL, thanks

**Review by awly — APPROVED** [review]

File: lib/httplib/httplib.go [hunk 5]
```diff
@@ -20,6 +20,7 @@ package httplib
 
 import (
```
Comment (awly) [inline lib/httplib/httplib.go:21]: missing `net` import

**Review by r0mant — APPROVED** [review]

**Pull request merged**